src/fidesops/ops/service/strategy_factory.py

```diff
@@ -0,0 +1,90 @@
+import logging
+from typing import Any, Callable, Dict, Type, ValuesView
+
+from pydantic import ValidationError
+
+from fidesops.ops.common_exceptions import NoSuchStrategyException
+from fidesops.ops.common_exceptions import ValidationError as FidesopsValidationError
+from fidesops.ops.service.strategy import Strategy
+
+logger = logging.getLogger(__name__)
+
+
+class StrategyFactory:
+    def __init__(self) -> None:
+        self.registry: Dict[str, Type[Strategy]] = {}
+        self.valid_strategies: str = ""
+
+    def register(self) -> Callable[[Type[Strategy]], Type[Strategy]]:
+        def wrapper(
+            strategy_class: Type[Strategy],
+        ) -> Type[Strategy]:
+
+            _validate_strategy_class(strategy_class)
+
+            name = strategy_class.name
+            logger.debug(
+                ("Registering new strategy '%s' under name '%s'", strategy_class, name)
+            )
+
+            if name in self.registry:
+                logger.warning(
+                    (
+                        "Strategy with name '%s' already exists. It previously referred to class '%s', but will now refer to '%s'",
+                        name,
+                        self.registry[name],
+                        strategy_class,
+                    )
+                )
+
+            self.registry[name] = strategy_class
+            self.valid_strategies = ", ".join(self.registry.keys())
+            return self.registry[name]
```

Comment:
Instead of this register decorator,  you might explore Metaclasses, or the `__subclasses__` method that returns all child classes.  I see you mentioned these as alternatives in the original issue.   I might lean on something more built-in, the register decorator adds an extra step and makes me think on first glance that it's magically going to add the strategy to the registry with its usage alone, when the strategy still needs to be imported to be registered.